### PR TITLE
Implement default policy configuration values

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#19826|

## Objective

This pull request introduces the default value when a minimal configuration valid is provided for the vulnerability detection module and indexer. Also, we fix and improve the validation logic.

## Description
- Implemented validation logic to verify the content of the configuration object.
- Added error-handling mechanisms to handle invalid or incomplete configurations.
- Included unit tests to verify the accuracy of the validation process.

## Quality Assurance
A new test has been included to verify the correct behavior and backward compatibility of this solution proposal.

<details><summary>Output testing 🟢 </summary>

```bash
[==========] Running 10 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 10 tests from PolicyManagerTest
[ RUN      ] PolicyManagerTest.invalidConfigurationNoIndexer
[       OK ] PolicyManagerTest.invalidConfigurationNoIndexer (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationNoVulnerabilityDetection
[       OK ] PolicyManagerTest.invalidConfigurationNoVulnerabilityDetection (0 ms)
[ RUN      ] PolicyManagerTest.getUpdaterConfiguration
[       OK ] PolicyManagerTest.getUpdaterConfiguration (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationIDWithVD
[       OK ] PolicyManagerTest.invalidConfigurationIDWithVD (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationURLFeed
[       OK ] PolicyManagerTest.invalidConfigurationURLFeed (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationNegativeTime
[       OK ] PolicyManagerTest.invalidConfigurationNegativeTime (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationIDWithTypo
[       OK ] PolicyManagerTest.invalidConfigurationIDWithTypo (0 ms)
[ RUN      ] PolicyManagerTest.invalidConfigurationVDWithTypo
[       OK ] PolicyManagerTest.invalidConfigurationVDWithTypo (0 ms)
[ RUN      ] PolicyManagerTest.validConfigurationCheckParameters
[       OK ] PolicyManagerTest.validConfigurationCheckParameters (0 ms)
[ RUN      ] PolicyManagerTest.validConfigurationDefaultValues
[       OK ] PolicyManagerTest.validConfigurationDefaultValues (0 ms)
[----------] 10 tests from PolicyManagerTest (1 ms total)

[----------] Global test environment tear-down
[==========] 10 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 10 tests.
```

</details>

Regarding memory, no new memory allocations were introduced in this PR using `raw_pointers`.

<details><summary>ScanBuild report</summary>

Command:

```bash
scan-build-15 -enable-checker alpha.core.CastSize -enable-checker alpha.core.CastToStruct -enable-checker alpha.core.Conversion -enable-checker alpha.core.IdenticalExpr -enable-checker alpha.core.SizeofPtr -enable-checker alpha.core.TestAfterDivZero --exclude external make TARGET=server DEBUG=1 -j$(nproc)
```

```bash

Done building server

scan-build: No bugs found.

```

</details>

<details><summary>Clang-Format</summary>

```bash
---------------------------------
PASSED: Clang-format check passed
---------------------------------
```
</details>

## Exploratory tests
Not apply

### Check summary
- [x] Style and quality of SCA.
- [x] Documentation clear.
- [x] Functionality optimizated.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/tests/policyManager/policyManager_test.cpp [hunk 1]
```diff
@@ -17,65 +17,121 @@ void PolicyManagerTest::SetUp()
...
+        "vulnerability-detection":
+        {
+          "dummy": true
+        }
+      }
+    )"_json;
 
-    EXPECT_TRUE(m_policyManager->isVulnerabilityDetectionEnabled());
-    EXPECT_TRUE(m_policyManager->isIndexerEnabled());
+    ASSERT_NO_THROW(m_policyManager->loadConfiguration(configJson));
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/tests/policyManager/policyManager_test.cpp:32]: The method `loadConfiguration()` simply copies a variable and performs no further operations.
It isn't testing anything.

File: src/wazuh_modules/vulnerability_scanner/tests/policyManager/policyManager_test.cpp [hunk 2]
```diff
@@ -17,65 +17,121 @@ void PolicyManagerTest::SetUp()
...
+    const auto& configJson = R"(
+      {
+        "indexer":
+        {
+          "dummy": true
+        }
+      }
+    )"_json;
+
+    ASSERT_NO_THROW(m_policyManager->loadConfiguration(configJson));
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/tests/policyManager/policyManager_test.cpp:47]: The method `loadConfiguration()` simply copies a variable and performs no further operations.
It isn't testing anything.

**Review by pereyra-m — COMMENTED** [review]
Gj!

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 3]
```diff
@@ -85,12 +74,15 @@ class PolicyManager final : public Singleton<PolicyManager>
...
         {
-            m_configuration["indexer"] = nlohmann::json::object();
-            m_configuration["indexer"]["servers"] = nlohmann::json::array();
-            m_configuration["indexer"]["servers"].push_back("http://localhost:9200");
-            m_configuration["indexer"]["databasePath"] = "queue/indexer_vd";
+            m_configuration["indexer"]["hosts"] = nlohmann::json::array();
+            m_configuration["indexer"]["hosts"].push_back("http://localhost:9200");
+        }
+
+        if (!m_configuration["vulnerability-detection"].contains("feed-update-interval"))
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:83]: when vulnerability-detection is not present, the module does not start.

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 4]
```diff
@@ -246,44 +173,69 @@ class PolicyManager final : public Singleton<PolicyManager>
...
+     *
+     * This function checks if the configuration settings are valid and conform
+     * to the expected format and values.
+     *
+     */
 
-        if (idObj.contains("ssl") && idObj["ssl"].contains("key"))
+    void validateConfiguration()
+    {
+        if (m_configuration.contains("vulnerability-detection") && m_configuration.contains("indexer"))
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:235]: yes, probably when "indexer" is not present.

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Pull request merged**